third_party/musl/crt/patch.diff

```diff
@@ -933,6 +933,35 @@ index eeae1d0a..7e13faa8 100644
  #endif
  }
 +weak_alias (sendmmsg, __sendmmsg);
+diff --git a/src/process/fork.c b/src/process/fork.c
+index fb42478a..44959ae8 100644
+--- a/src/process/fork.c
++++ b/src/process/fork.c
+@@ -22,7 +22,15 @@ pid_t fork(void)
+ #else
+ 	ret = __syscall(SYS_clone, SIGCHLD, 0);
+ #endif
+-	if (!ret) {
++
++#if 0
++       // In Linux, the child process can safely modify the libc internal variables as below
++       // since (v)fork creates a new memory space for the child; any changes done by the child
++       // does not affect the parent.
++       // However, in mystikos, after (v)fork, the child still shares the same libc/crt as the parent.
++       // Therefore, it is not safe to modify the libc state. It is only after execve that the child
++       // gets its own crt/libc.
++       if (!ret) {
+ 		pthread_t self = __pthread_self();
```

Comment:
Setting pthread state in the child should work if __pthread_self() fetches the right thread descriptor.

Reply:
Turns out this line is needed. LTP tests use (sometimes) use fork. Retaining the logic to set the pthread_t members allow these tests to continue to pass. 

Although I'm not sure how robust the fork implementation is in the first place. Earlier, after the fork, the system was incorrectly put in a "single-threaded" mode and for some tests, a different code path was taken.